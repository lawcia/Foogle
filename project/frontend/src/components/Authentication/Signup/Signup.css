.Signup__form {
  width: 300px;
  display: flex;
  flex-direction: column;
  padding: 16px;
  margin: 16px 0;
}

.Signup__input {
  margin-top: 4px;
}

.Signup__row {
  margin-bottom: 16px;
}

.Signup__input--error {
  border: 1px solid #630700;
}

.Signup__logo {
  width: 200px;
}

.Signup__label {
  font-weight: 600;
}

.Signup__text--error {
  color: #630700;
  padding: 4px;
}