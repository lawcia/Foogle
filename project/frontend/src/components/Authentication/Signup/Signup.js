import React from "react";
import PropTypes from "prop-types";

const Signup = ({
  handleSubmit,
  handleChange,
  user,
  usernameError,
  emailError,
}) => {
  return (
    <div className="form-div">
      <form className="user-form" onSubmit={handleSubmit}>
        <input
          className="input input-m"
          onChange={handleChange}
          value={user.username}
          type="text"
          name="username"
          placeholder="Create a username"
          minLength="3"
          maxLength="25"
          required
        />
        {usernameError && (
          <p className="align-text error-msg">{usernameError}</p>
        )}
        <input
          className="input input-m"
          onChange={handleChange}
          value={user.email}
          type="email"
          name="email"
          placeholder="Enter your email address"
          maxLength="50"
          required
        />
        {emailError && <p className="align-text error-msg">{emailError}</p>}
        <input
          className="input input-m"
          onChange={handleChange}
          value={user.password}
          type="password"
          name="password"
          placeholder="Create a password"
          minLength="6"
          maxLength="20"
          required
        />
        <input
          className="input input-m"
          onChange={handleChange}
          value={user.password2}
          type="password"
          name="password2"
          placeholder="Please repeat your password"
          required
        />
        <button className="button is-success input-m">Sign up</button>
      </form>
    </div>
  );
};

Signup.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
  usernameError: PropTypes.object.isRequired,
  emailError: PropTypes.object.isRequired
}

export default Signup;
