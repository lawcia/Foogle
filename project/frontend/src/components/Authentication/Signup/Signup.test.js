import React from "react";
import Signup from "./Signup";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("<Signup />", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<Signup 
    handleSubmit={jest.fn()}
    handleChange={jest.fn()}
    user={{username: '', email: '', password: '', password2: ''}}
    usernameError={null}
    emailError={null}
    passwordError={null}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it("should show username error message", () => {
    const message = "this username has been taken";
    const wrapper = shallow(<Signup
                             handleSubmit={jest.fn()}
                             handleChange={jest.fn()}
                             user={{username: '', email: '', password: '', password2: ''}}
                             usernameError={{message: message}}
                             emailError={null}
                             passwordError={null}
                            />);
    expect(wrapper.find(".Signup___text--error").length).toEqual(1);
    expect(wrapper.find(".Signup___text--error").text()).toEqual(message);
    expect(wrapper.find("input.Signup__input--error").length).toEqual(1);
  })
  it("should show email error message", () => {
    const message = "this email has been taken";
    const wrapper = shallow(<Signup
                             handleSubmit={jest.fn()}
                             handleChange={jest.fn()}
                             user={{username: '', email: '', password: '', password2: ''}}
                             usernameError={null}
                             emailError={{message: message}}
                             passwordError={null}
                            />);
    expect(wrapper.find(".Signup___text--error").length).toEqual(1);
    expect(wrapper.find(".Signup___text--error").text()).toEqual(message);
    expect(wrapper.find("input.Signup__input--error").length).toEqual(1);
  })

  it("should show email error message", () => {
    const message = "this password is too short";
    const wrapper = shallow(<Signup
                             handleSubmit={jest.fn()}
                             handleChange={jest.fn()}
                             user={{username: '', email: '', password: '', password2: ''}}
                             usernameError={null}
                             emailError={null}
                             passwordError={{message: message}}
                            />);
    expect(wrapper.find(".Signup___text--error").length).toEqual(1);
    expect(wrapper.find(".Signup___text--error").text()).toEqual(message);
    expect(wrapper.find("input.Signup__input--error").length).toEqual(1);
  })

  it("should not have error messages", () => {
    const wrapper = shallow(<Signup
                             handleSubmit={jest.fn()}
                             handleChange={jest.fn()}
                             user={{username: '', email: '', password: '', password2: ''}}
                             usernameError={null}
                             emailError={null}
                             passwordError={null}
                            />);
    expect(wrapper.find(".Signup___text--error").length).toEqual(0);
    expect(wrapper.find("input.Signup__input--error").length).toEqual(0);
  })
})