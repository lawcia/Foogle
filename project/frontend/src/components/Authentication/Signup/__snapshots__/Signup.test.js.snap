// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> should match snapshot 1`] = `
<div
  className="form-div"
>
  <form
    className="Signup__form"
    onSubmit={[MockFunction]}
  >
    <div
      className="Signup__row"
    >
      <label
        className="Signup__label"
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="input input-m Signup__input null"
        id="username"
        maxLength="25"
        minLength="3"
        name="username"
        onChange={[MockFunction]}
        type="text"
        value=""
      />
    </div>
    <div
      className="Signup__row"
    >
      <label
        className="Signup__label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="input input-m Signup__input null"
        id="email"
        maxLength="50"
        name="email"
        onChange={[MockFunction]}
        type="email"
        value=""
      />
    </div>
    <div
      className="Signup__row"
    >
      <label
        className="Signup__label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="input input-m Signup__input null"
        id="password"
        maxLength="20"
        minLength="6"
        name="password"
        onChange={[MockFunction]}
        type="password"
        value=""
      />
    </div>
    <div
      className="Signup__row"
    >
      <label
        className="Signup__label"
        htmlFor="password2"
      >
        Repeat password
      </label>
      <input
        className="input input-m Signup__input null"
        id="password2"
        name="password2"
        onChange={[MockFunction]}
        type="password"
        value=""
      />
    </div>
    <button
      className="button is-primary blue"
    >
      Sign up
    </button>
  </form>
</div>
`;
